icu4j/main/classes/core/src/com/ibm/icu/text/PluralRules.java

```diff
@@ -1298,13 +1258,14 @@ private static void checkDecimal(SampleType sampleType2, FixedDecimal sample) {
         public String toString() {
             StringBuilder b = new StringBuilder("@").append(sampleType.toString().toLowerCase(Locale.ENGLISH));
             boolean first = true;
-            for (FixedDecimalRange item : samples) {
+            for (DecimalQuantitySamplesRange item : samples) {
                 if (first) {
                     first = false;
                 } else {
                     b.append(",");
                 }
-                b.append(' ').append(item);
+                String dqStr = item.toString();
```

Comment:
Nit (optional): the general best practice is to give `item` a function where it can append to the StringBuilder directly, rather than converting it to a `String` first